bin/seed_experience_index.py

```diff
@@ -0,0 +1,292 @@
+                course.name,
+                course.iri,
+            )
+            experiences += [
+                ExperienceFactory.build(
+                    iri=course.iri,
+                    title=course.name,
+                    structure="atomic",
+                    aggregation_level=AggregationLevel.THREE,
+                )
+            ]
+            courses.add(course.iri)
+
+        if action.iri in actions:
+            continue
+
+        # Create a new experience for this action
+        logger.debug(
+            "Will create Experience for action: %s with iri: %s",
+            action.name,
+            action.iri,
+        )
+        experiences += [
+            ExperienceFactory.build(
+                iri=action.iri,
+                title=action.name,
+                structure="atomic",
+                aggregation_level=AggregationLevel.ONE,
+            )
+        ]
+        actions.add(action.iri)
+        course_action_iris += [(course.iri, action.iri)]
+
+    logger.info(
+        "Will create %d experiences (%d courses + %d actions)",
+        len(experiences),
+        len(courses),
+        len(actions),
+    )
+    logger.debug("Courses IRIs: %s", courses)
+    logger.debug("Actions IRIs: %s", actions)
+    # Add the new experiences to the Experience Index
+    batch_size = 1000
+    try:
+        for i, batch in enumerate(batched(experiences, batch_size)):
+            logger.info("Inserting batch %d (size: %d)", i, batch_size)
+            session.add_all(batch)
+            session.flush()
+        session.commit()
+    except SQLAlchemyError:
+        logging.error("Could not create Experiences. Rolling back database session.")
+        session.rollback()
+        session.close()
+        raise
+
+    logger.info("Created %d experiences", len(experiences))
+
+    logger.info("Fetching related course/action IDs")
+    # Get courses and actions from the database
+    db_courses = pd.read_sql(
+        sql=select(Experience)
+        .where(Experience.aggregation_level == AggregationLevel.THREE)
+        .order_by(Experience.iri.asc()),
+        con=engine,
+    )
+    db_actions = pd.read_sql(
+        sql=select(Experience)
+        .where(Experience.aggregation_level == AggregationLevel.ONE)
+        .order_by(Experience.iri.asc()),
+        con=engine,
+    )
+    # Create a dataframe from course/action IRIs
+    course_action_iris = pd.DataFrame(course_action_iris, columns=["course", "action"])
+    # Iterate over course/action dataframe by chunks
+    course_action_relations = pd.DataFrame(columns=["course_id", "action_id"])
+    chunk_size = 1000
+    for idx in range(0, course_action_iris.shape[0], chunk_size):
+        logger.info(
+            "Create course/action IDs dataframe for chunk %d (size: %d)",
+            idx,
+            chunk_size,
+        )
+        # Add course IDs from database given course IRIs
+        tmp_relations = pd.merge(
```

Comment:
I don't get why you have to create a temporary dataframe. I think you can concat directly based on a condition on IRIS, no ?

Reply:
As merging two big dataframes could be time-consuming, splitting those merge events in smaller chunks is more efficient.

Reply:
alright